test/integration/customUrlRest.spec.ts

```diff
@@ -0,0 +1,16 @@
+import CustomUrlRestClient, { CustomUrlRestClientRestClient } from "./generated/customUrlRest/src";
+import { assert } from 'chai';
+
+describe("CustomRest Endpoint", () => {
+  let client: CustomUrlRestClientRestClient;
+  let clientOptions: any;
+  beforeEach(() => {
+    clientOptions = { allowInsecureConnection: true };
+    client = CustomUrlRestClient("host:3000", clientOptions);
+  });
+
+  it("should return 200", async () => {
+    const result = await client.path("/customuri", 'local').get();
```

Comment:
According to the conversation in https://github.com/Azure/autorest.typescript/issues/1057

I'm expecting this call to look like this
```typescript
// Internally as defined in the swagger, the base url has parameters in this case the base url is: `{accountName}{suffix}`, 
// where suffix is a global parameter that we pass when building the client as above, and `accountName` is a method 
// parameter that we'll require as part of the operation options.
await client.path("/customuri").get({pathParameters: { accountName: "local"} })
```

For this we need to:

1. Add a new `pathParameters` property to `RequestParameters` in @azure-rest/core-client`
2. Update logic in `@azure-rest/core-client` to use `pathParameters` when building the url.
3. Generate the parameter types to include `accountName` as a pathParameter

For the changes in (2) you'd need to add logic before this loop https://github.com/Azure/azure-sdk-for-js/blob/a8b7f42993733e43a3dbaee89e66393df1b8bfa0/sdk/core/core-client-rest/src/urlHelpers.ts#L27

In which you loop the object options.pathParameters (added as part of (1)) to replace any occurrences of its keys in the URL with the value.

Reply:
Got it. Thanks for info and instructions